department-of-veterans-affairs/vets-website
# Certificate of Eligibility | Update to web components

## Original Ticket

[#47111](https://github.com/department-of-veterans-affairs/va.gov-team/issues/47111)

## URL of change

- `/housing-assistance/home-loans/request-coe-form-26-1880/introduction`
- `/housing-assistance/home-loans/check-coe-status/your-coe/`

## Background

COE was still using React Components that need to be updated to use the equivalent Web Component.

While modifying code, a problem with the UX was discovered (see https://github.com/department-of-veterans-affairs/va.gov-team/issues/47469) which needs to be addressed, but this PR did include some changes to include already submitted files within the success message

After changing `FileInput` to `va-file-input`, updating the unit tests proved to be difficult due to testing-library/react not supporting testing the shadow DOM, making it difficult (impossible?) to click/trigger an upload. So use of the `FileInput` React component was restored

## Steps to test

1. Pull in this branch locally, or test in the review instance
2. Log in with user 228 (Mark Webb)
3. Go to the first URL of change
4. Check that the web components are in place
5. Go to the second URL of change
6. Check that the web components are in place - you won't be able to test the uploader without manipulating the code

## Code changes

### Form app
- `OMBInfo` → `va-omb-info`
- `va-telephone` → add `tty` prop
- Download link → `va-link`

### Status app
- `TextInput` → `va-text-input`
- `Select` → `va-select`
- Download link → `va-link` & remove static file size value. Pull file extension from file name (`mimeType`).
- Fix date within download links from `MMDDYYY` to `MM-DD-YYYY` to make it more readable
- Link to start COE form → action link
  <details><summary>COE status page</summary>
  <img width="731" alt="COE status with proper download links with icons and a blue action link to start the form" src="https://user-images.githubusercontent.com/136959/192335683-1b20b99d-77f1-4636-ae9d-35d5a90534d8.png">
  </details>
- Include submitted files within the success message
  <details><summary>Upload success</summary>
  <img width="714" alt="upload success message with 2 files that were just uploaded listed within" src="https://user-images.githubusercontent.com/136959/192336179-a11ca243-c8fc-4816-a0d9-10896ed7be19.png">
  </details>

## How was your code tested

Updated unit & e2e tests

## Acceptance criteria
- [x] Update to use web components
- [x] Fix download link formatting
- [x] Minor a11y & UX fixes

## Definition of done
